`make gardener-down` fails at deletion of `garden` `Project`

**How to categorize this issue?**
<!--
Please select area, kind, and priority for this issue. This helps the community categorizing it.
Replace below TODOs or exchange the existing identifiers with those that fit best in your opinion.
If multiple identifiers make sense you can also state the commands multiple times, e.g.
  /area control-plane
  /area auto-scaling
  ...

"/area" identifiers:     audit-logging|auto-scaling|backup|certification|control-plane-migration|control-plane|cost|delivery|dev-productivity|disaster-recovery|documentation|high-availability|logging|metering|monitoring|networking|open-source|ops-productivity|os|performance|quality|robustness|scalability|security|storage|testing|usability|user-management
"/kind" identifiers:     api-change|bug|cleanup|discussion|enhancement|epic|impediment|poc|post-mortem|question|regression|task|technical-debt|test
-->
/area dev-productivity
/kind bug

**What happened**:
Calling `make gardener-down` fails with the following error:
```
$ make gardener-down 
hack/tools/bin/kubectl delete seed local --ignore-not-found --wait --timeout 5m
seed.core.gardener.cloud "local" deleted
hack/tools/bin/kubectl delete seed local2 --ignore-not-found --wait --timeout 5m
hack/tools/bin/skaffold delete -m provider-local,gardenlet
Cleaning up...
release "gardener-extension-provider-local-registration" uninstalled
release "gardener-gardenlet" uninstalled
hack/tools/bin/kubectl delete validatingwebhookconfiguration/validate-namespace-deletion --ignore-not-found
validatingwebhookconfiguration.admissionregistration.k8s.io "validate-namespace-deletion" deleted
hack/tools/bin/kubectl annotate project local confirmation.gardener.cloud/deletion=true
project.core.gardener.cloud/local annotated
hack/tools/bin/skaffold delete -m local-env
Cleaning up...
 - secret "bootstrap-token-07401c" deleted
 - namespace "garden-local" deleted
 - secret "backup-local" deleted
 - secret "local" deleted
 - cloudprofile.core.gardener.cloud "local" deleted
 - controllerdeployment.core.gardener.cloud "networking-calico" deleted
 - controllerregistration.core.gardener.cloud "networking-calico" deleted
 - project.core.gardener.cloud "local" deleted
 - secretbinding.core.gardener.cloud "local" deleted
 - secretbinding.core.gardener.cloud "local" deleted
 - secret "seed-local" deleted
 - Error from server (Forbidden): error when deleting "STDIN": projects.core.gardener.cloud "garden" is forbidden: must have a "confirmation.gardener.cloud/deletion" annotation to delete
kubectl delete: exit status 1
make: *** [gardener-down] Error 1
```

**What you expected to happen**:
`make gardener-down to complete successfully`

**How to reproduce it (as minimally and precisely as possible)**:
1. `make kind-up`
2. `make gardener-up`
3. `make gardener-down`

**Anything else we need to know?**:
The `garden` `Project` was added to the `kustomize` configuration here: https://github.com/gardener/gardener/blob/master/example/provider-local/garden/base/project.yaml with #6293

Perhaps we could add `make gardener-down` to our e2e tests, maybe as a last step in the [ci-e2e-kind.sh](https://github.com/gardener/gardener/blob/master/hack/ci-e2e-kind.sh) script to prevent future errors.

**Environment**:

- Gardener version: 1da39002ccd90e8b3dc08763dcf41708e48d907f
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration: 
- Others:

--- a/Makefile
+++ b/Makefile
@@ -316,7 +316,7 @@ gardener-down: $(SKAFFOLD) $(HELM) $(KUBECTL)
 	$(KUBECTL) delete seed local2 --ignore-not-found --wait --timeout 5m
 	$(SKAFFOLD) delete -m provider-local,gardenlet
 	$(KUBECTL) delete validatingwebhookconfiguration/validate-namespace-deletion --ignore-not-found
-	$(KUBECTL) annotate project local confirmation.gardener.cloud/deletion=true
+	$(KUBECTL) annotate project local garden confirmation.gardener.cloud/deletion=true
 	$(SKAFFOLD) delete -m local-env
 	$(SKAFFOLD) delete -m etcd,controlplane
 	@# workaround for https://github.com/gardener/gardener/issues/5164

--- a/hack/ci-e2e-kind-migration.sh
+++ b/hack/ci-e2e-kind-migration.sh
@@ -32,6 +32,7 @@ trap "
     export_events_for_kind 'gardener-local'; export_events_for_shoots )
   ( export KUBECONFIG=$PWD/example/gardener-local/kind2/kubeconfig; export_logs 'gardener-local2';
     export_events_for_kind 'gardener-local2' )
+  ( make kind-down; make kind2-down )
 " EXIT
 
 make gardener-up
@@ -40,5 +41,6 @@ make gardenlet-kind2-up
 # run test
 make test-e2e-local-migration
 
+# test teardown
 make gardener-down
 make gardenlet-kind2-down

--- a/hack/ci-e2e-kind-migration.sh
+++ b/hack/ci-e2e-kind-migration.sh
@@ -39,3 +39,6 @@ make gardenlet-kind2-up
 
 # run test
 make test-e2e-local-migration
+
+make gardener-down
+make gardenlet-kind2-down

--- a/hack/ci-e2e-kind.sh
+++ b/hack/ci-e2e-kind.sh
@@ -35,3 +35,5 @@ make gardener-up
 
 # run test
 make test-e2e-local
+
+make gardener-down

--- a/hack/ci-e2e-kind.sh
+++ b/hack/ci-e2e-kind.sh
@@ -29,6 +29,7 @@ make kind-up
 trap "
   ( export KUBECONFIG=$PWD/example/gardener-local/kind/kubeconfig; export_logs 'gardener-local';
     export_events_for_kind 'gardener-local'; export_events_for_shoots )
+  ( make kind-down )
 " EXIT
 
 make gardener-up
@@ -36,4 +37,5 @@ make gardener-up
 # run test
 make test-e2e-local
 
+# test teardown
 make gardener-down

